pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -322,7 +326,18 @@ func Convert_v1_CloudInitNoCloudSource_To_api_Disk(source *v1.CloudInitNoCloudSo
 		return fmt.Errorf("device %s is of type lun. Not compatible with a file based disk", disk.Alias.Name)
 	}
 
-	disk.Source.File = fmt.Sprintf("%s/%s", cloudinit.GetDomainBasePath(c.VirtualMachine.Name, c.VirtualMachine.Namespace), cloudinit.NoCloudFile)
+	disk.Source.File = cloudinit.GetNoCloudIsoFilePath(c.VirtualMachine.Name, c.VirtualMachine.Namespace)
+	disk.Type = "file"
+	disk.Driver.Type = "raw"
+	return nil
+}
+
+func Convert_v1_CloudInitConfigDriveSource_To_api_Disk(source *v1.CloudInitConfigDriveSource, disk *Disk, c *ConverterContext) error {
```

Comment:
`source` arg is not used here (I know it's the same as `Convert_v1_CloudInitNoCloudSource_To_api_Disk`), can we remove it?

I think it makes sense to merge this function with `Convert_v1_CloudInitNoCloudSource_To_api_Disk` and prepare a generic one for volume-based cloud-init, both functions do the same thing. It would require a small additional refactoring for `cloudinit.GetConfigDriveIsoFilePath` and `cloudinit.GetNoCloudIsoFilePath`, we would have to do the same thing and prepare `cloudinit.GetISOFilePath` with a datasource type argument. It prevents having the same function copies.

Any thoughts?

Reply:
Sounds good to me. I think we should prevent function copies as much as possible. I'll see if I can merge these functions.

Reply:
@mfranczy I tried to incorporate your suggestions. Would be nice if you could have another look.

Reply:
Looks good, just please remove commented code, https://github.com/kubevirt/kubevirt/pull/2120/commits/6cd5a10d1dcb5c9a79e32151b57750b2cb6c5b1b#diff-caee6d384498fde52fde15891ac547aaR340
and let's wait for CI

Reply:
Haha, I removed it but then forgot to save the file in my editor :P
pushed it again without the commented code.

Reply:
Thanks, let's wait for tests result now.